quirks/password-rules.json

```
@@ -266,6 +269,9 @@
     "hrblock.com": {
         "password-rules": "minlength: 8; required: lower; required: upper; required: digit; required: [$#%!];"
     },
+    "hsbc.com.my": {
+        "password-rules": "minlength: 8; maxlength: 30; required: lower, digit; allowed: lower, upper, digit, [!$*-.=?@_'];"
```

A couple notes about this rule:

* Any character classes that are `required` don't need to be specified in `allowed`.
* The screenshot of the site's password rule information states "It must contain at least 1 letter and 1 number". The rules `required` property will need to be `required: lower, upper; required: digit` to express that.
* The `-` character must be the first character in a custom character class.

That means the rule will end up being:

```
minlength: 8; maxlength: 30; required: lower, upper; required: digit; allowed: [-!$*.=?@_'];
```

---

Implemented your suggestion:
```suggestion
        "password-rules": "minlength: 8; maxlength: 30; required: lower, digit; allowed: lower, upper, digit, [!$*-.=?@_'];"
```